File: conformance/bundles/04-tables-and-docstrings/echo.steps.ts
import { defineState } from '@oselvar/var'

const { sensor } = defineState(() => ({}))

// The doc string is this sensor's only slot, so it is returned bare; the core
// compares it against the input (compareDocString); equal content passes.
sensor('I echo the following:', (_ctx, doc: string) => doc)

File: conformance/bundles/04-tables-and-docstrings/echo.steps.cs
// C# sibling of echo.steps.ts / .rs (bundle 04-tables-and-docstrings).
using Varar;
using Varar.Core;

namespace Varar.Corpus.B04;

public static class EchoSteps
{
    public static void Register(Steps s)
    {
        // The doc string is this sensor's only slot, so it is returned bare; the core compares it
        // against the input (equal passes).
        s.Sensor("I echo the following:", (state, doc) => doc);
    }

    public static Value State() => Value.Null;
}
